### PR TITLE
Added phpstan checks

### PR DESCRIPTION
**Pull request by core23**
## Subject

Improve code quality by running phpstan 🚀 

I am targeting this branch, because this is BC, but should only be done for the next major version.

## Changelog

```markdown
### Removed
- Remove `PolyfillControllerTrait`
- Remove `ModelChoiceList`
```

### CODE REVIEW

**Comment by core23** [comment]
PHPStan is running :)

**Comment by OskarStark** [comment]
Nice, can you check Travis?

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -46,22 +44,11 @@
...
-// NEXT_MAJOR: Remove the polyfill and inherit from \Symfony\Bundle\FrameworkBundle\Controller\Controller again
-if (!trait_exists(ControllerTrait::class)) {
-    require_once __DIR__.'/PolyfillControllerTrait.php';
-}
-
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-class CRUDController implements ContainerAwareInterface
+class CRUDController extends Controller
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]: As a bonus/out of scope: avoid extending this by using DI

**Comment by greg0ire** [comment]
docs and lint are in allowed failures, in the build… something is very wrong :sweat_smile:

**Comment by core23** [comment]
> docs and lint are in allowed failures, in the build… something is very wrong 😅

Can you create a `merge master` PR @greg0ire ?

**Comment by greg0ire** [comment]
See https://github.com/sonata-project/SonataAdminBundle/pull/5769

**Comment by greg0ire** [comment]
Thanks @core23 !

**Pull request merged**